### PR TITLE
Ensure vendor directory is up-to-date as part of CI

### PR DESCRIPTION
**Pull request by nmiodice**
## All Submissions:
-------------------------------------
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?
-------------------------------------
The CI does not currently validate that the `vendor` directory is up to date. With the changes here, it will. In addition, the documentation is also updated to reflect this, as is the PR template.

Issue Number: https://github.com/microsoft/terraform-provider-azuredevops/issues/311

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?
-------------------------------------
- [NO]

## Does this introduce a breaking change?
-------------------------------------
- [NO]

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-* [YES/NO] I have updated the documentation accordingly.
-* [YES/NO/NA] I have added tests to cover my changes.
-* [YES/NO/NA] All new and existing tests passed.
-* [YES/NO/NA] My code follows the code style of this project.
-* [YES/NO/NA] I ran lint checks locally prior to submission.
-* [YES/NO] Have you checked to ensure there aren't other open PRs for the same update/change?
-
-## What is the current behavior?
+* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
+* [ ] I have updated the documentation accordingly.
```
Comment (EliiseS) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: Now THIS is a great update! Who needs shmancy vendor checks when we have real check boxes!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
 
-Issue Number: N/A
+Issue Number:
 
-
-## What is the new behavior?
+## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?
 -------------------------------------
-<!-- Please describe the behavior or changes that are being added by this PR. -->
+- [YES/NO]
```
Comment (EliiseS) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Maybe checkbox too? I hate changing these texts sooo much

Comment (tmeckel) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: Can we get rid of the lines under the headings as well... Sorry I couldn't resist 😂😂😂😂

Comment (nmiodice) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done :)

**Pull request merged**